### PR TITLE
Automated cherry pick of #11252

### PR DESCRIPTION
**Pull request by brendandburns**
Fixes https://github.com/kubernetes/kubernetes/issues/12737 in 1.0.X

@fabioy @nikhiljindal

### CODE REVIEW

**Comment by nikhiljindal** [comment]
Change LGTM, thanks!
I will recommend updating the PR title to say "Automated cherry pick of #" as we do for automated cherrypicks.

File: pkg/apiserver/apiserver_test.go [hunk 1]
```diff
@@ -2330,7 +2330,7 @@ func TestCreateTimeout(t *testing.T) {
 
 	simple := &Simple{Other: "foo"}
 	data, _ := codec.Encode(simple)
-	itemOut := expectApiStatus(t, "POST", server.URL+"/api/version/foo?timeout=4ms", data, apierrs.StatusServerTimeout)
+	itemOut := expectApiStatus(t, "POST", server.URL+"/api/version/namespaces/default/foo?timeout=4ms", data, apierrs.StatusServerTimeout)
```
Comment (fabioy) [inline pkg/apiserver/apiserver_test.go]: Could you add a test that verifies the fix (i.e. test with an actual empty namespace)?

File: api/swagger-spec/v1.json [hunk 2]
```diff
@@ -10916,43 +10503,6 @@
...
-        "message": "OK",
-        "responseModel": "v1.Service"
-       }
-      ],
-      "produces": [
-       "application/json"
-      ],
-      "consumes": [
-       "*/*"
-      ]
```
Comment (fabioy) [inline api/swagger-spec/v1.json]: Swagger change seems ok, but I'll leave it for someone who knows it better.

**Comment by zmerlynn** [comment]
Cross link #11252

File: api/swagger-spec/v1.json (hunk 2 above)
Comment (nikhiljindal) [inline api/swagger-spec/v1.json]: Swagger spec changes LGTM

File: pkg/apiserver/apiserver_test.go (hunk 1 above)
Comment (nikhiljindal) [inline pkg/apiserver/apiserver_test.go]: Note that we should add the test in master and not in this cherrypick PR.

**Comment by nikhiljindal** [comment]
LGTM

File: pkg/apiserver/apiserver_test.go (hunk 1 above)
Comment (fabioy) [inline pkg/apiserver/apiserver_test.go]: SGTM.

**Comment by fabioy** [comment]
LGTM.

**Pull request merged**